inst/include/model_simulation.hpp

```diff
@@ -239,87 +239,129 @@ void run_h_art_initiation(int hiv_step,
   const auto art = pars.base.art;
   const auto dt = pars.base.options.dt;
   const auto hIDX_15PLUS = pars.base.options.hIDX_15PLUS;
+
   for (int g = 0; g < ss.NS; ++g) {
+
     intermediate.base.Xart_15plus = 0.0;
+
+    intermediate.base.artelig_hm.setZero();
     intermediate.base.Xartelig_15plus = 0.0;
+
+    intermediate.base.expect_mort_artelig_hm.setZero();
     intermediate.base.expect_mort_artelig15plus = 0.0;
+    
     for (int ha = hIDX_15PLUS; ha < ss.hAG; ++ha) {
       for (int hm = intermediate.base.everARTelig_idx; hm < ss.hDS; ++hm) {
+	
         if (hm >= intermediate.base.anyelig_idx) {
           // TODO: Implement special population ART eligibility
           real_type prop_elig = 1.0;
-          intermediate.base.artelig_hahm(hm, ha - hIDX_15PLUS) =
-              prop_elig * state_next.base.h_hiv_adult(hm, ha, g);
-          intermediate.base.Xartelig_15plus += intermediate.base.artelig_hahm(
-              hm, ha - hIDX_15PLUS);
-          intermediate.base.expect_mort_artelig15plus +=
-              natural_history.cd4_mortality(hm, ha, g) *
-              intermediate.base.artelig_hahm(hm, ha - hIDX_15PLUS);
+
+	  real_type tmp_artelig = prop_elig * state_next.base.h_hiv_adult(hm, ha, g);
```

Comment:
Add to the intermediate data object rather than define here.

Reply:
Regarding this and the other similar comments — my understanding is the `intermediates` object is to avoid re-declaring large temporary blocks of memory on the heap.  This and the other ones flagged are single bits on the stack. They will have negligible overhead.

Is there any benefit to putting these in 'intermediate data'? 

@r-ash maybe weigh in as well.

Reply:
Oh yes, I did not realise that. But yes I think no need to add these into intermediate data as they are just stack allocated, sorry if I've misled you on that before Maggie. I think also good to keep that here as keeps the scope of this intermediate data a bit smaller, which is always a positive. I wonder if they are still stack allocated in TMB run where real_type will be one of the TMB types?

Reply:
Thanks @r-ash. I'm pretty sure that they will still be for the TMB types, but even if not, it's known at compile time should be well handled.